builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -244,6 +249,25 @@ func resourceCloudStackInstanceUpdate(d *schema.ResourceData, meta interface{})
 
 	name := d.Get("name").(string)
 
+	if d.HasChange("name") {
+		log.Printf("[DEBUG] Name for %s changed to %s, starting update", d.Id(), name)
+
+		// Create a new parameter struct
+		p := cs.VirtualMachine.NewUpdateVirtualMachineParams(d.Id())
+
+		// Set the new name
+		p.SetName(d.Get("name").(string))
```

Comment:
We already have the `name` variable declared on line 250, so this can just be `p.SetName(name)`